routes/auth.js

```diff
@@ -0,0 +1,90 @@
+import express from 'express';
+const router = express.Router();
+import mongoose from "mongoose";
+import User from "../models/user.js";
+import bcrypt from "bcryptjs";
+import jwt from "jsonwebtoken";
+import requireLogin from "../middleware/requiredLogin.js"
+
+
+router.get("/protected", requireLogin, (req, res) => {
+    res.send("Hello user to protected routes")
+})
+
+
+const JWT_SECRET = "ayush_Token";
+
+
+router.post("/signup", (req, res) => {
+    const { name, email, password, pic } = req.body;
+
+    if (!email || !name || !password) {
+        return res.status(401).json({ error: "Please fill all fields completely" })
+    }
+
+    User.findOne({ email: email })
+        .then((saveduser) => {
+            if (saveduser) {
+                return res.status(422).json({ error: "user already exists" })
+            }
```

Comment:
The code inside the routes file is huge. Make a controller folder and put all the CRUD operations there and just refer the functions name in routes files when you do get, post, delete, update. Do this for all the files of routes i.e. auth, user, post.